### DISCUSSION TITLE
Total counts

### DISCUSSION THREAD

**Discussion by CodeTile** [enisn/AutoFilterer#72] [General]
When using applying filtering and paging, how can I get the Total record count and the Total page count?

I could not find a reference to this in the documentation.

**Comment by enisn** [comment]
You can use `ApplyFilterWithoutPagination` to get all count with that filter but without applying pagination:
```cs
var queryResult = await dbContext.Books.ApplyFilter(filter).ToListAsync();
var totalCount = await dbContext.Books.ApplyFilterWithoutPagination(filter).CountAsync();
```

**Reply by CodeTile** [reply]
Thank you

**Comment by CodeTile** [comment]
Works a treat.
